### ISSUE TITLE
Simple Sites - Site Editor loads default home template indefinitely when custom domain is primary

### ISSUE THREAD

**Issue by villanovachile** [Automattic/wp-calypso#76105]
### Quick summary

When a simple site has a custom domain set as primary, navigating to Appearance > Editor will show a loading screen indefinitely as shown below:

<img width="1442" alt="image" src="https://user-images.githubusercontent.com/65082164/233798950-37e3629d-ae68-4f1b-b045-2dd9fc8b745e.png">

Direct links to the templates and template parts work as expected. I believe this is due to the iframe removal as discussed here: p7DVsv-h8i-p2

As noted there: `The URL in their browser will no longer be of the form https://wordpress.com/site-editor/SITE_SLUG but instead https://SITE_SLUG/wp-admin/site-editor.php`

Because of this, the wp-admin site_slug will always be the simple site WPCOM subdomain, and this seems to be causing an issue when there aren't any URL query strings proceeding `wp-admin/site-editor.php`

Console log errors seem to confirm this with CORS related issues due to the mismatch of the `SITE_SLUG` (thanks @ash1eygrace for pointing this out!):

```
Access to fetch at 'https://atomic11119823y72hia.blog/?_wp-find-template=true' from origin
'https://simple11111.wordpress.com/' has been blocked by CORS policy: 
No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque 
response serves your needs, set the request's mode to 'no-cors' to fetch the  resource with 
CORS disabled.
```

As such, this isn't reproducible on Atomic nor self hosted and is isolated to simple sites that are using a custom domain as the primary address.

Workaround is easy, all a user needs to do is navigate to the template or template part they wish to view. 

### Steps to reproduce

1. Assign a custom domain as the primary address
2. Navigate to Appearance > Editor
3. The preview of the site will load indefinitely

### What you expected to happen

I expected a preview of the site to load

### What actually happened

A loading screen displayed indefinitely

### Impact

Most (> 50%)

### Available workarounds?

Yes, easy to implement

### Platform (Simple and/or Atomic)

Simple

**Comment by villanovachile** [comment]
Related: https://github.com/Automattic/wp-calypso/issues/75577#issue-1663049892

**Comment by dlaurisse** [comment]
Another ticket here: 6226029-zen

**Comment by villanovachile** [comment]
Report here 6226186-zen

**Comment by villanovachile** [comment]
Another report 6226579-zen

**Comment by cuemarie** [comment]
FYI HEs this has been escalated here, and folks are looking into it! p1682186205979559-slack-C02FMH4G8

**Comment by cuemarie** [comment]
📌 SCRUBBING : RESULT - *Replicated / Could Not Replicate / Uncertain*
- Tested on Simple – **Replicated**

📌 FINDINGS/SCREENSHOTS/VIDEO
- I am able to replicate only when the homepage is set to `--Default--` - when a static homepage is selected, the Site Editor loads without issue.

#### Screenshots/Recordings

https://user-images.githubusercontent.com/27249804/233811943-6c944eae-cdad-48cd-953a-84459370d9f0.mp4

Firefox tracked the console error:

```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://mqteststarter.com/?_wp-find-template=true. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing). Status code: 200.
```

📌 ACTIONS
- Assigned to Cylon

**Comment by cuemarie** [comment]
Note on impact - Issues https://github.com/Automattic/themes/issues/6966 and https://github.com/Automattic/wp-calypso/issues/67434 have seen a high volume of interactions added to them in the last 48 hours (~15 each), most of which I suspect belong over here.

I've pinged the reporting HEs to ask them to take a look and share here if the shoe fits.

**Comment by isocialtish** [comment]
Reported in 6226828-zen got browser info here: https://www.whatismybrowser.com/w/4GS78H8

**Comment by essleeung** [comment]
Found another one here in 41165892-hc
Follow up here: 6227007-zd-woothemes

**Comment by dlaurisse** [comment]
Another tickets here: **6225715-zen** and **6223661-zen**

**Comment by laragee6** [comment]
6224027-zen
6225582-zen
6196511-zen
6225685-zen
6258837-zen
6265600-zen

**Comment by laragee6** [comment]
6225582- zen

**Comment by isocialtish** [comment]
Another report at 6223632-zen

**Comment by isocialtish** [comment]
6225058-zen

**Comment by isocialtish** [comment]
6223671-zen

**Comment by isocialtish** [comment]
6226472-zen

**Comment by dlaurisse** [comment]
another tickets: 
**6225736-zen**
**6226568-zen**

**Comment by 74MR47** [comment]
I can replicate the issue only when I set the homepage to `--Default-- `and use a custom domain as the site's primary site address. However, when I select a static homepage or use the free `.wordpress.com` as the site's primary site address, the Site Editor loads without any problems.

ZD-6226224
ZD-6226346
ZD-6226504
ZD-6226947
ZD-6227092
ZD-6226876
ZD-6227044

**Comment by jp-imagines** [comment]
Another in 6226040-zd-woothemes.

**Comment by trixxmanaty** [comment]
same here: 6226683-zen

**Comment by philnick206** [comment]
More here:
6227916-zd-woothemes
6227811-zd-woothemes
6227798-zd-woothemes
6227711-zd-woothemes
6227643-zd-woothemes
6227258-zd-woothemes
6227256-zd-woothemes

**Comment by dlaurisse** [comment]
another ticket: **6227207-zen**

**Comment by rw-ye** [comment]
Another report: 6228507-zd-woothemes

**Comment by adelineyaw** [comment]
Another report: 6225966-zen

**Comment by pak-cia** [comment]
+1 - 6227557-zen

**Comment by thesacredpath** [comment]
And another one: 42856357-hc

I'm on Firefox latest and could reproduce.

**Comment by Greatdane** [comment]
Marking this high priority due to the number of users affected.

**Comment by trixxmanaty** [comment]
another one here: 6227206-zen

**Comment by worldomonation** [comment]
Revert PR has been merged, awaiting deploy to production.

**Comment by worldomonation** [comment]
https://github.com/Automattic/wp-calypso/pull/76106

**Comment by jorpdesigns** [comment]
Reported in 6223137-zd-woothemes

**Comment by ariel-maidana** [comment]
6229395-zd-woothemes

**Comment by philnick206** [comment]
6228425-zd-woothemes
6228500-zd-woothemes
6228459-zd-woothemes
6228543-zd-woothemes

**Comment by mgozdis** [comment]
39595815-hc

**Comment by 74MR47** [comment]
👉 ZD-6228137 & ZD-6228763

**Comment by csabarakasz** [comment]
6229601-zd-woothemes

https://[SITE URL].wordpress.com/wp-admin/site-editor.php **doesn't load**.
https://wordpress.com/site-editor/[PRIMARY_DOMAIN] **does load**.

**Comment by csabarakasz** [comment]
42806666-hc

**Comment by upwardmomentum84** [comment]
36749608-hc

**Comment by cuemarie** [comment]
Looks like this is resolved - not sure if that's from https://github.com/Automattic/wp-calypso/pull/76106 (since some interactions were listed after that point) or GB 15.7, but closing this as resolved!

https://user-images.githubusercontent.com/27249804/236568585-06e15831-26ed-4a82-8a09-5d65970543b2.mp4

For new reports, please make sure the details of the issue match the situation in the original report exactly (Simple Site + Custom Domain + Default Homepage + CORS related error in console logs). If they're different, please report as a  new issue.

**Comment by happychait** [comment]
I just encountered this while working on my test site as in the following screenshot: 

 `Access to XMLHttpRequest at 'https://chaittestpress.com/?_wp-find-template=true&_locale=user' from origin 'https://chaittestpress.wordpress.com' has been blocked by CORS policy: Request header field x-wp-nonce is not allowed by Access-Control-Allow-Headers in preflight response.`

![CleanShot 2023-05-23 at 12 38 16@2x](https://github.com/Automattic/wp-calypso/assets/1519466/abe1a69e-d7a2-484c-8adb-99b79989e9e2)

Video of Site Editor taking forever to load when the custom domain is primary.

https://github.com/Automattic/wp-calypso/assets/1519466/daf29ccf-898e-44ba-bd7b-f94285aa6933

**Comment by worldomonation** [comment]
I tried to reproduce this issue in my environment.

- os: macOS Ventura
- browser: firefox nightly 115
- site has custom domain (.blog) set as primary.
- theme: heiwa

I was unable to reproduce the issue. Please see attached video.

@happychait Seeing that this issue has many people in the loop and was completed some weeks ago, my suggestion would be to open a new issue.


https://github.com/Automattic/wp-calypso/assets/6549265/6c5ed394-24bc-4d61-8e47-7855308411b0

**Comment by csabarakasz** [comment]
6474373-zd-woothemes

**Comment by Gustavo-Hilario** [comment]
I've reproduced the issue multiple times when switching to user's sites. I don't know if something specific with support sessions is causing this problem. It also happens when the free domain name is the primary one. I'm leaving a note here in case someone else comes across this same issue.

p9cu9o-3SY-p2

**Comment by mrfoxtalbot** [comment]
I can still reproduce this on 7560768-zd-a8c both with the default and the custom domain.

@worldomonation, @happychait, I am going to re-open this issue. If this is being tracked elsewhere, our you think we should open a separate report, please let me know.

Thank you!

**Comment by annezazu** [comment]
I have a simple site with a custom domain name and can't replicate this. Going to close this out as a result since there haven't been additional reports since January and I cannot replicate this.
